File: 1_twosum/c#/twoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int[] res={0,1};
        int arr_len = nums.Length;
        Dictionary<int, int> val_idx = new Dictionary<int, int>();
        for(int i = 0; i < arr_len;i++){
            if(val_idx.ContainsKey(nums[i])){
                res[0] = val_idx[nums[i]];
                res[1] = i;
                break;
            }
            else{
                val_idx[target-nums[i]] = i; 
            }
        }
        return res;
    }
}

File: 1_twosum/c++/twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        int arr_len = nums.size();
        map<int,int> tab;
        for(int i = 0; i < arr_len; i++){
            if(tab.count(nums[i]) != 0){
                cout<<nums[i]<<endl;
                res.push_back(tab[nums[i]]);
                res.push_back(i);
            }
            else{
                tab[target-nums[i]] = i;
            }
        }
        return res;
    }
};